soc/nordic/nrf54h/pm_s2ram.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/arch/common/pm_s2ram.h>
+#include <zephyr/linker/sections.h>
+#include <zephyr/sys/util.h>
+#include <hal/nrf_resetinfo.h>
+#include <pm_s2ram.h>
```

Comment:
this should be included as `"pm_s2ram.h"` to make it clear it's a local private header